plutus-tx-plugin/test/Budget/allCheap.plc.golden

```diff
@@ -2,66 +2,33 @@ letrec data (List :: * -> *) a | Nil_match where
-      CConsMultiplicativeMonoid
-      : (\a -> a -> a -> a) a -> a -> MultiplicativeMonoid a
-in fFoldableNil_cfoldMap
-     {(\a -> a) Bool} {integer} (let !v : MultiplicativeMonoid Bool
-                                       = CConsMultiplicativeMonoid
-                                           {Bool} (\(l : Bool) (r : Bool) ->
-                                                     Bool_match
-                                                       l {all dead. Bool}
-                                                       (/\dead -> r) (/\dead ->
-                                                                        False)
-                                                       {all dead. dead}) True
-                                 in CConsMonoid
-                                      {(\a -> a) Bool}
-                                      (\(eta : (\a -> a) Bool) (eta
-                                        : (\a -> a) Bool) ->
-                                         MultiplicativeMonoid_match
-                                           {Bool} v {(\a -> a -> a -> a) Bool}
-                                           (\(v : (\a -> a -> a -> a) Bool) (v
-                                             : Bool) ->
-                                              v) eta eta)
-                                      (MultiplicativeMonoid_match
-                                         {Bool} v {Bool}
-                                         (\(v : (\a -> a -> a -> a) Bool) (v
-                                           : Bool) ->
-                                            v))) (\(v : integer) ->
-                                                    ifThenElse
-                                                      {Bool}
-                                                      (lessThanEqualsInteger
-                                                         1 v) False True)
-     ((let a = List integer in \(c : integer -> a -> a) (n : a) ->
-                                 c
-                                   1 (c
-                                        2 (c
-                                             3 (c
+in
+letrec !go : List integer -> Bool
+         = \(ds : List integer) ->
+             Nil_match
+               {integer} ds {all dead. Bool} (/\dead -> True)
+               (\(x : integer) (xs : List integer) ->
+                  /\dead ->
+                    let !acc : Bool = go xs in Bool_match
+                                                 (ifThenElse
+                                                    {Bool}
+                                                    (lessThanEqualsInteger 1 x)
+                                                    False True) {all dead. Bool}
+                                                 (/\dead -> acc) (/\dead ->
+                                                                    False)
+                                                 {all dead. dead})
+               {all dead. dead}
```

Comment:
So everything works just fine except GHC makes a thunk for `go xs` and pulls it out of `ifThenElse`. And we interpret it a strict binding.